### PR TITLE
Python 3.10 improvements

### PR DESCRIPTION
**Pull request by hugsy**
## Description

Follow-up PR for #1133 but purely esthetic, no functional changes

Cleans up the code to make it more in comformance with Python 3.10 capabilities.

Replaces:
 - type `Union` with `|`
 - type `Dict`/`Set`/`List`/`Tuple` with `dict`/`set`/`list`/`tuple`
 - `Optional` with ` | None`
 - `if:` ... `elif`... -> `match`

Fixes: 
 - `gdb.events.*Event` -> `gdb.*Event`
 - `GefHeapManager.malloc_align_address` had its `ceil` lambda moved as a proper (nested) function

Lint:
 - `except:` -> `except Exception:`
 - removed a bunch of unused f-strings
 - `if not .. in ..` -> `if .. not in ..`
 - some short named variables renamed

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -110,6 +110,8 @@ def tearDown(self) -> None:
...
     @property
     def gdb_version(self) -> Tuple[int, int]:
-        res = [int(d) for d in re.search(r"(\d+)\D(\d+)", self._gdb.VERSION).groups()]
-        assert len(res) >= 2
-        return tuple(res)
+        res = re.search(r"(\d+)\D(\d+)", self._gdb.VERSION)
+        assert res
+        groups = [int(d) for d in res.groups()]
+        assert len(groups) == 2
+        return (groups[0], groups[1])
```
Comment (Grazfather) [inline tests/base.py:117]:
```suggestion
        return groups[0], groups[1]
```

File: gef.py [hunk 2]
```diff
@@ -9911,25 +9914,26 @@ def __init__(self) -> None:
         gef.config["gef.libc_version"] = GefSetting("", str, "Specify libc version when auto-detection fails")
         gef.config["gef.main_arena_offset"] = GefSetting("", str, "Offset from libc base address to main_arena symbol (int or hex). Set to empty string to disable.")
         gef.config["gef.propagate_debug_exception"] = GefSetting(False, bool, "If true, when debug mode is enabled, Python exceptions will be propagated all the way.")
+        gef.config["gef.extra_python_package_paths"] = GefSetting("", str, "Semi-colon separate list of extra paths to include for python packages")
 
-        self.commands : Dict[str, GenericCommand] = collections.OrderedDict()
-        self.functions : Dict[str, GenericFunction] = collections.OrderedDict()
-        self.missing: Dict[str, Exception] = {}
+        self.commands : dict[str, GenericCommand] = collections.OrderedDict()
```
Comment (Grazfather) [inline gef.py:9919]: don't we not need ordered dicts anymore?

File: gef.py [hunk 3]
```diff
@@ -10694,7 +10698,7 @@ def reset_caches(self) -> None:
                 if not hasattr(obj, "cache_clear"):
                     continue
                 obj.cache_clear()
-            except: # we're reseting the cache here, we don't care if (or which) exception triggers
+            except Exception: # we're reseting the cache here, we don't care if (or which) exception triggers
```
Comment (Grazfather) [inline gef.py:10701]:
```suggestion
            except Exception: # we're resetting the cache here, we don't care if (or which) exception triggers
```

File: gef.py [hunk 4]
```diff
@@ -11063,7 +11071,7 @@ def selected_arena(self, value: GlibcArena) -> None:
         return
 
     @property
-    def arenas(self) -> Union[List, Iterator[GlibcArena]]:
+    def arenas(self) -> list | Iterator[GlibcArena]:
```
Comment (Grazfather) [inline gef.py:11074]: If you want to use `|` I think there are many more cases where we can get rid of `Union`

File: gef.py [hunk 5]
```diff
@@ -11710,10 +11724,30 @@ def reset_caches(self) -> None:
...
 
 def target_remote_posthook():
-    if gef.session.remote_initializing:
-        return
+    conn = gdb.selected_inferior().connection
+    # if here, the connection should be established
+    if not isinstance(conn, gdb.RemoteTargetConnection):
+        raise EnvironmentError("Failed to create a remote session")
+
+    # if gef.session.remote_initializing:
```
Comment (Grazfather) [inline gef.py:11732]: remove?

File: gef.py [hunk 6]
```diff
@@ -11710,10 +11724,30 @@ def reset_caches(self) -> None:
...
+    #     dbg(f"remote session already initializing: {gef.session}")
+    #     return
+
+    print(f"{conn.description=}")
```
Comment (Grazfather) [inline gef.py:11736]: gef print?

File: gef.py [hunk 7]
```diff
@@ -11771,6 +11794,14 @@ def target_remote_posthook():
         # reload settings
         gdb.execute("gef restore")
 
+    # extend `sys.path` for venv compat
+    if gef.config["gef.extra_python_package_paths"]:
+        for path in map(pathlib.Path, gef.config["gef.extra_python_package_paths"].split(";")):
+            if not path.exists():
+                warn(f"Skipping invalid directory {path}")
+                continue
+            sys.path.extend(str(path.absolute()))
```
Comment (Grazfather) [inline gef.py:11803]: is path.absolutely not already a string?

File: gef.py [hunk 8]
```diff
@@ -11789,32 +11820,16 @@ def target_remote_posthook():
...
-            gdb.execute(hook.format("remote"))
-            gdb.execute(hook.format("extended-remote"))
-        else:
-            errmsg = ("Using `target remote` does not work, use `gef-remote` "
-                      f"instead. You can toggle `{disable_tr_overwrite_setting}` "
-                      "if this is not desired.")
-            hook = f"""pi if calling_function() != "connect": err("{errmsg}")"""
-            gdb.execute(f"define target hook-remote\n{hook}\nend")
-            gdb.execute(f"define target hook-extended-remote\n{hook}\nend")
+    # Register a post-hook for `target remote` that initialize the remote session
```
Comment (Grazfather) [inline gef.py:11823]:
```suggestion
    # Register a post-hook for `target remote` that initializes the remote session
```

**Comment by hugsy** [comment]
I removed all comments because the PR is still draft and having those messes up how VSCode displays them directly into my code.

**Review by ValekoZ — APPROVED** [review]
LGTM

**Pull request merged**